### PR TITLE
aioredis, Nominatim cache

### PR DESCRIPTION
**Pull request by jonathanspw**
Updates redis to use aioredis w/hiredis.

Implement caching for Nominatim data via redis.

### CODE REVIEW

File: src/backend/db/db_engine.py [hunk 1]
```diff
@@ -72,12 +72,4 @@ class RedisEngine:
...
-        if not cls.__instance:
-            cls.__instance = Redis(
-                host=REDIS_HOST,
-                port=REDIS_PORT,
-                db=REDIS_DB,
-                charset="utf-8",
-                decode_responses=True
-            )
-        return cls.__instance
+        return aioredis.from_url("redis://%s:%s/%s" % (REDIS_HOST, REDIS_PORT, REDIS_DB))
```
Comment (soksanichenko) [inline src/backend/db/db_engine.py:75]: f-string is better variant and shorter than current

File: src/backend/api/redis.py [hunk 2]
```diff
@@ -21,15 +23,17 @@
...
         key: AnyStr,
 ) -> Optional[List[MirrorData]]:
     """
     Get a cached list of mirrors for specified IP
     """
     key = str(key)
     redis_engine = RedisEngine.get_instance()
-    mirrors_string = redis_engine.get(key)
+    mirrors_string = await redis_engine.get(key)
+    await redis_engine.close()
```
Comment (soksanichenko) [inline src/backend/api/redis.py:35]: Can we move the two lines to a separate function?

File: src/backend/api/redis.py [hunk 3]
```diff
@@ -48,20 +52,66 @@ def set_mirrors_to_cache(
...
+    key = str(key)
+    redis_engine = RedisEngine.get_instance()
+    coords = await redis_engine.get(key)
+    await redis_engine.close()
+    await redis_engine.connection_pool.disconnect()
+    if coords:
+        return json.loads(coords)
+
+
+async def set_geolocation_to_cache(key: AnyStr, coords: Tuple) -> None:
```
Comment (soksanichenko) [inline src/backend/api/redis.py:74]: FYI: python3.9 supports built-in types for typing)

**Review by soksanichenko — CHANGES_REQUESTED** [review]

File: src/backend/api/redis.py (hunk 2 above)
Comment (jonathanspw) [reply src/backend/api/redis.py:35]: The engine closing?  I don't see why...

File: src/backend/api/redis.py (hunk 3 above)
Comment (jonathanspw) [reply src/backend/api/redis.py]: Thanks :)

**Review by soksanichenko — APPROVED** [review]

**Pull request merged**